### PR TITLE
Avoid wrapping prediction dataloader twice on TPU

### PR DESCRIPTION
**Pull request by Liyang90**
Avoid wrapping dataloader with `MpDeviceLoader` more than once. Add `batch_sampler` back to the dataloader.

## What does this PR do?

Fixes [#16572](https://github.com/Lightning-AI/lightning/issues/16572)

When doing prediction on TPU with:
```
trainer=Trainer(accelerator='tpu', devices=8)
trainer.predict(model=model, dataloaders=dm)
```
the code would crash because the prediction dataloader is wrapped twice with `MpDeviceLoader`, once in [`DataConnector._reset_eval_dataloader()`](https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/trainer/connectors/data_connector.py#L278) and once in [`PredictionLoop.advance()`](https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/loops/dataloader/prediction_loop.py#L112).

To avoid double wrapping, a checking is added in `TPUSpawnStrategy.process_dataloader()`. 

The `batch_sampler` is also added to the wrapped dataloader to avoid warning in [prediction_epoch_loop](https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/loops/epoch/prediction_epoch_loop.py#L182).

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Yes
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by carmocca** [comment]
CI status: blocked by https://github.com/Lightning-AI/lightning/pull/16613
